### PR TITLE
Wip solution

### PR DESCRIPTION
**Pull request by azat-co**
CC @AnnJohn @vicfriedman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,268 @@
...
+## Objectives
+
+1. Write a simple non-npm module with module.exports = {}
+1. Import your module from another file
+1. Invoke a method from the module
+
+
+## Introduction
+
+In this lab, you'll master how to create, export and import a object module, i.e., a static module.
```
Comment (victhevenot) [inline README.md]: the intro needs to frame the "domain" of this lab. we're have customer data, but what kind of customers? for what? why do we need to separate out a module here? The more we can connect the labs to work the students can inherently understand, the more motivated they will be. Just having them create modules without a purpose can feel like theyre just doing busy work.

File: README.md [hunk 2]
```diff
@@ -1 +1,268 @@
...
+1. Create a file `module.js` with `module.exports = {}`
+2. Copy customer data object from the snippet below into `module.js` and export the data
+3. Create a file `main.js` 
+4. Import your object in `main.js` with `require()`
+5. Print an email of the second customer from `main.js`
+6. Keep `main.js` and `module.js` in the same folder
+6. Run `$ npm test` to verify
+
+```js
+[
```
Comment (victhevenot) [inline README.md:26]: the data  object provided here is LONG and hard to digest. can we shorten this to just a few properties? its hard to even find where an email is for the second customer

File: README.md [hunk 3]
```diff
@@ -1 +1,268 @@
...
+    "favoriteFruit": "apple"
+  }
+]
+```
+
+### Extra Info
+
+The data is an array. In JavaScript/Node, arrays are objects with extra properties so the pattern `module.exports = {}` will work the same for arrays: `module.exports = []`.
+
+Arrays are 0-based so the second element will have an index of 1.
```
Comment (victhevenot) [inline README.md]: students know all about arrays already. can delete this

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    console.log('stdout: ' + stdout)
+    console.log('stderr: ' + stderr)
+    t.equal(stderr, '')
+    if (error !== null) {
+      console.log('exec error: ' + error)
+    }
+    stdout = stdout.replace('\n','')
+    t.equal(stdout, 'scott.wilkerson@hopeli.me')
+  })
+})
```
Comment (victhevenot) [inline test/index.js:9]: these tests are pretty hard to read. similar comments to other labs - but students don't know what a child process is or how to interpret `cp.exec`....

is there a way to simplify the tests by changing the work -  to have them define functions that return specific data

**Comment by azat-co** [comment]
updated @vicfriedman @AnnJohn

**Comment by victhevenot** [comment]
ready for you @pletcher  + wip-master branch

File: README.md (hunk 3 above)
Comment (pletcher) [inline README.md]: Ah, oops -- I made a comment on the same line asking for more explanation in the other PR. I'd side with @vicfriedman on this one.

**Comment by azat-co** [comment]
updated @vicfriedman @AnnJohn @pletcher

**Comment by victhevenot** [comment]
@azat-co - remove note about arrays being 0-indexed. students definitely already know this.

**Comment by pletcher** [comment]
Sorry about that -- that was my fault re: arrays.

File: README.md [hunk 5]
```diff
@@ -1 +1,88 @@
-# node-modules-object-lab
\ No newline at end of file
+# Using Object Module
+
+## Objectives
+
+1. Write a simple non-npm module with module.exports = {}
```
Comment (pletcher) [inline README.md]: I think we can drop the emphasis on non-npm vs. npm modules -- it ends up just being confusing for the student, I think.

File: README.md [hunk 6]
```diff
@@ -1 +1,88 @@
...
+In this lab, you'll master how to create, export and import an object module, i.e., a static module.
+
+## Instructions
+
+1. Create a file `module.js` with `module.exports = {}`
+2. Copy customer data object from the `customers.txt` (sample is below) into `module.js` and export the data
+3. Create a file `main.js`
+4. Import your object in `main.js` with `require()`
+5. Print an email of the second customer from `main.js`
+6. Keep `main.js` and `module.js` in the same folder
```
Comment (pletcher) [inline README.md]: I don't think we need to specify this.

File: README.md [hunk 7]
```diff
@@ -1 +1,88 @@
-# node-modules-object-lab
\ No newline at end of file
+# Using Object Module
+
+## Objectives
+
+1. Write a simple non-npm module with module.exports = {}
+1. Import your module from another file
+1. Invoke a method from the module
```
Comment (pletcher) [inline README.md]: It seems that students don't actually accomplish this objective. They just `console.log` the second email address.

File: README.md [hunk 8]
```diff
@@ -1 +1,88 @@
...
+## Introduction
+
+We have some customer data for testing our application before we implement the real database or REST API. It's a bad idea to store this data in the code, so your task is to organize the data as a separate file. This way it will be easier to switch from this customer data to some other source later.
+
+In this lab, you'll master how to create, export and import an object module, i.e., a static module.
+
+## Instructions
+
+1. Create a file `module.js` with `module.exports = {}`
+2. Copy customer data object from the `customers.txt` (sample is below) into `module.js` and export the data
```
Comment (pletcher) [inline README.md]: This feels super janky. I know we don't cover `require('something.json')` for another few lessons, but since the emphasis of this lessons is on exporting static data from one module and importing it in another, can we reconfigure it so that the student instead needs to add the appropriate `module.exports` to an existing file of customer data?

We could, for example, change `customers.txt` to `customer.js` and task them with exporting the data (but not telling them exactly how, since this is a lab).

File: README.md [hunk 9]
```diff
@@ -1 +1,88 @@
...
+  {...},
+  {...}
+]
+```
+
+### Extra Info
+
+The data is an array. In JavaScript/Node, arrays are objects with extra properties so the pattern `module.exports = {}` will work the same for arrays: `module.exports = []`.
+
+Tip: Arrays in JavaScript are 0-based, so the second element of an array will have an index of 1.
```
Comment (pletcher) [inline README.md]: Let's remove this, per @vicfriedman's comment

**Comment by pletcher** [comment]
@azat-co back at you! :)

**Comment by azat-co** [comment]
updates @pletcher @vicfriedman @AnnJohn
